crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -421,8 +422,8 @@ pub struct SlidingSyncRoom {
     pub name: Option<String>,
 
     /// The avatar of the room.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub avatar: Option<OwnedMxcUri>,
+    #[serde(skip_serializing_if = "JsOption::is_undefined")]
```

Comment:
```suggestion
    #[serde(default, skip_serializing_if = "JsOption::is_undefined")]
```

It's important that non-`Option` fields have a `default` attribute if an absent field should not cause an error with serde's derives.